src/common/common.h

```diff
@@ -43,13 +43,24 @@ extern "C" {
  * This is a temporary workaround until a better error
  * handling strategy is developed.
  */
-#define OQS_OPENSSL_GUARD(x)    \
-    do {                        \
-        if( 1 != (x) ) {        \
+#if defined(OQS_USE_OPENSSL) && !defined(OPENSSL_NO_STDIO)
+#define OQS_OPENSSL_GUARD(x)                                                           \
+    do {                                                                               \
+        if( 1 != (x) ) {                                                               \
             fprintf(stderr, "Error return value from OpenSSL API: %d. Exiting.\n", x); \
-            exit(EXIT_FAILURE); \
-        }                       \
+            OSSL_FUNC(ERR_print_errors_fp)(stderr);                                    \
+            exit(EXIT_FAILURE);                                                        \
+        }                                                                              \
     } while (0)
+#else
+#define OQS_OPENSSL_GUARD(x)                                                           \
```

Comment:
Is the following #define triggered if "OQS_USE_OPENSSL" is NOT set? Wouldn't seem quite right.

Reply:
Good catch. I've refactored the directives so that the guard is defined if and only if `OQS_USE_OPENSSL` is defined.